ccm_web/server/src/lti/lti.service.ts

```diff
@@ -49,29 +48,30 @@ export class LTIService implements BeforeApplicationShutdown {
     provider.onConnect(async (token: IdToken, req: Request, res: Response) => {
       logger.debug(`The LTI launch was successful! User info: ${JSON.stringify(token.userInfo)}`)
       const customLTIVariables = token.platformContext.custom
-      if (customLTIVariables === null) {
-        return res.json({ lti_error: 'LTI launch is missing custom attributes, please add it in LTI configutation in Canvas' })
+      if (customLTIVariables?.login_id === undefined) {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({ lti_error: 'LTI launch is missing custom attributes, please add it in LTI configutation in Canvas' })
       }
       const loginId = customLTIVariables.login_id as string
-      const [record, created] = await this.userService.upsertUser({
-        firstName: token.userInfo.given_name,
-        lastName: token.userInfo.family_name,
-        email: token.userInfo.email,
-        loginId: loginId
-      })
-      /* created variable will return non null value for Mysql, but the Return type on upsert method is Promise<[User, boolean|null] >
+      try {
+        const [record, created] = await this.userService.upsertUser({
+          firstName: token.userInfo.given_name,
+          lastName: token.userInfo.family_name,
+          email: token.userInfo.email,
+          loginId: loginId
+        })
+
+        /* created variable will return non null value for Mysql, but the Return type on upsert method is Promise<[User, boolean|null] >
       so Typescript is mandating a null check. So here the null is changed to false to escape the type validation errors */
-      logger.debug(
-        `User record for ${record.loginId} was ${
-          (created ?? false) ? 'created' : 'updated'
-        } in 'user' table`
-      )
-      // if the user inserting into DB failed don't want to proceed further
-      if (!(record instanceof User)) {
-        logger.error(`something went wrong in creating user: ${loginId}`)
+        logger.info(
+          `User ${record.loginId} is ${
+            (created ?? false) ? 'created' : 'updated'
+          } in 'user' table`
+        )
+        return provider.redirect(res, '/')
+      } catch (e) {
```

Comment:
it will error out like this `- error - something went wrong in creating user: undefined with error SequelizeDatabaseError due to Field 'loginId' doesn't have a default value`

It will throw `SequelizeDatabaseError` when insertion fails. Looks like There is no easy way to mention multiple exceptions in catch but check it manually like in [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch-blocks).

In our case we don't care what type of error to throw different messages around it.